framework/src/executioners/FixedPointSolve.C

```diff
@@ -198,18 +205,23 @@ FixedPointSolve::solve()
 
   Real current_dt = _problem.dt();
 
-  _fixed_point_timestep_begin_norm.clear();
-  _fixed_point_timestep_end_norm.clear();
-  _fixed_point_timestep_begin_norm.resize(_max_fixed_point_its);
-  _fixed_point_timestep_end_norm.resize(_max_fixed_point_its);
+  _fixed_point_begin_norm.clear();
+  _fixed_point_end_norm.clear();
+  _fixed_point_begin_norm.resize(_max_fixed_point_its);
+  _fixed_point_end_norm.resize(_max_fixed_point_its);
 
   bool converged = true;
 
   // need to back up multi-apps even when not doing fixed point iteration for recovering from failed
   // multiapp solve
   _problem.backupMultiApps(EXEC_MULTIAPP_FIXED_POINT_BEGIN);
-  _problem.backupMultiApps(EXEC_TIMESTEP_BEGIN);
-  _problem.backupMultiApps(EXEC_TIMESTEP_END);
+  if (_legacy_execute_on)
+  {
+    _problem.backupMultiApps(EXEC_TIMESTEP_BEGIN);
+    _problem.backupMultiApps(EXEC_TIMESTEP_END);
+  }
+  _problem.backupMultiApps(EXEC_FIXEDPOINT_BEGIN);
+  _problem.backupMultiApps(EXEC_FIXEDPOINT_END);
   _problem.backupMultiApps(EXEC_MULTIAPP_FIXED_POINT_END);
```

Comment:
you can see in these lines that you are adding a new flag that overlaps with an existing flag

Reply:
I am adding two flags `EXEC_FIXEDPOINT_BEGIN` and `EXEC_FIXEDPOINT_END` and move the existing two flags into the if statement. What do you mean *overlaps*?

Reply:
ok then the new schedule for `TIMESTEP_BEGIN` is the same as `EXEC_MULTIAPP_FIXED_POINT_BEGIN`

Reply:
In `Transient` executioner, yes because fixed point solve is the immediate inner solver object of the time stepper, but in general the two could be different. Maybe we want to remove the two `EXEC_MULTIAPP_FIXED_POINT_BEGIN` and `EXEC_MULTIAPP_FIXED_POINT_END`?

Reply:
no. What we could do is rename your new EXEC_FIXEDPOINT_BEGIN flag as EXEC_MULTIAPP_FIXED_POINT_BEGIN and decide to flip the execution schedule between TIMESTEP_BEGIN and  EXEC_MULTIAPP_FIXED_POINT_BEGIN

but this requires a careful discussion. This is making loosely coupled the default behavior in many cases

Reply:
this is going to fall onto the users more than the developers. If the app devs select the new behavior, every user's tightly coupled input will become a loosely coupled input (if using PPs & auxkernels for the kernel).
The fix is simple for sure.

Reply:
Although we could make pps and auxkernels default on fixedpoint_end, I think it is better to raise the awareness of users to ask them to add execute_on for those to be transferred during the fixed point iterations. We could add some checks in the Transfers to make sure quantities transferred during fixed point iterations are updated.

Reply:
Checks would be good. It's tough though because the lagging could be intentional.
I think we ll have to error when lagging is not intentional

Reply:
We could add the error check and a parameter to bypass the error in case lagging is intentional.